### DISCUSSION TITLE
`count_num_comparisons_from_blocking_rule` missing from new Linker API

### DISCUSSION THREAD

**Discussion by bkitej-rw** [moj-analytical-services/splink#2305] [Q&A] [answered]
### What happens?

In Splink 3, the main `Linker` objects had the method `count_num_comparisons_from_blocking_rule`. This seems to be absent from the Splink 4 API, though it is still referenced in [documentation](https://github.com/moj-analytical-services/splink/blob/d5ccd46e63e9bfadad844bd2e777bc4d8eeeafcd/docs/topic_guides/blocking/model_training.md?plain=1#L49), and an [unused helper function exists](https://github.com/moj-analytical-services/splink/blob/d5ccd46e63e9bfadad844bd2e777bc4d8eeeafcd/splink/internals/match_key_analysis.py#L2).

### To Reproduce

Inspect the `Linker` object

### OS:

MacOS 14.2.1

### Splink version:

4.0.0

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by RobinL** [comment, marked as answer]
See the [tutorial](https://moj-analytical-services.github.io/splink/demos/tutorials/03_Blocking.html) for the new way of doing blocking analysis.

```

from splink.blocking_analysis import count_comparisons_from_blocking_rule

db_api = DuckDBAPI()

br = block_on("substr(first_name, 1,1)", "surname")

counts = count_comparisons_from_blocking_rule(
    table_or_tables=df,
    blocking_rule=br,
    link_type="dedupe_only",
    db_api=db_api,
)

counts
```

Thanks for spotting the unused code, we'll remove that!

**Reply by bkitej-rw** [reply]
ah my bad, thank you!
